Alert modal to confirm lost progress from signing in on an in-progress form

U-DON opened the pull request:
## Description
Display a modal warning logged out users that they will lose form progress if they sign in.

~~This depends on changes from department-of-veterans-affairs/design-system#250. We need that fix for alert-styled modals to even remotely resemble the design from the mocks, as passing a status without the fix would result in a really broken-looking modal.~~ The Formation changes have been merged.

## Testing done
Local testing. Verified with HCA that there was no warning on the introduction page and that there was when in the middle of the actual form.

Checked to see that the modals from GIBCT landing page were not broken.

## Screenshots
> <img width="645" alt="screen shot 2019-02-19 at 12 03 48 pm" src="https://user-images.githubusercontent.com/1067024/53033423-000dd000-343f-11e9-8101-cedec7b927ed.png">

## Acceptance criteria
- [x] Logged out users should be warned that they will lose form progress if they sign in while filling out the form.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

U-DON committed:
First pass at modal for leaving forms to sign in.

U-DON committed:
Style tweaks for modals that are children of the header.

U-DON committed:
Changes to support updated modal close button.

U-DON committed:
Merge branch 'master' into vetsgovteam-16806

narin reviewed (commented):
Not sure if this is within the scope of this ticket, but if you have this modal open, and hit the back button in the browser, the modal stays open but the page content changes.

![ezgif-2-f3e9fb823936](https://user-images.githubusercontent.com/786704/53197394-f87d3100-35ce-11e9-84bd-01341e1b57e8.gif)

U-DON committed:
Merge branch 'master' into vetsgovteam-16806

U-DON committed:
Merge branch 'master' into vetsgovteam-16806

U-DON committed:
Updated content and design of modal.

U-DON committed:
Merged style rule that was too general into more specific rule.

U-DON committed:
Refactor to report login modal close event similarly to login modal open event.

U-DON committed:
Close login modals when navigating to different routes in an app.

U-DON committed:
Removed obsolete code.

U-DON committed:
Updated unit test.

U-DON committed:
Merge branch 'master' into vetsgovteam-16806

U-DON commented on src/platform/forms/save-in-progress/FormSignInModal.jsx:
```diff
@@ -0,0 +1,51 @@
+class FormSignInModal extends React.Component {
+  handleSignIn = () => {
+    this.props.onClose();
+    this.props.onSignIn();
```
This is a case where coupling the GA event with the action might be a problem... It seems like we might want to report a `login-link-restart-form` event here, but since it's opening the modal, by default, we would also be reporting a `login-link-clicked-cta` event.

U-DON replied:
The question is whether we are okay with both events firing.

U-DON replied:
This has been OK'd from a product perspective.

U-DON commented:
Forgot that this will depend on some Formation changes. Among other modifications, it will make it so that the header text of a modal doesn't wrap under the icon. department-of-veterans-affairs/veteran-facing-services-tools#8

U-DON committed:
Added more user nav reducer tests.

U-DON committed:
Added unit test helper to mock event listeners.

U-DON committed:
Simplified some logic.

U-DON committed:
Cleaned up mockEventListeners and added event object argument for simulate.

U-DON committed:
More comprehensive unit tests for user nav container.

U-DON committed:
Merge branch 'master' into vetsgovteam-16806

U-DON committed:
Associated GA events with actions in FormSignInModal.

U-DON committed:
Added FormSignInModal unit tests.

U-DON committed:
Minor change to user nav container unit test setup.

U-DON commented:
@narin: This is ready for another review.

## Updates:
* Included Formation updates for spacing tweaks, making sure the header text doesn't run under the icon, and general changes to make alerts and alert modals closely match.
* Test coverage. I have held off on coverage (in user-nav's Main.jsx) for binding links with the classes `login-required` or `sign-in-sign-up-modal-trigger` as I'm considering cleaning up that logic.
* Added GA events to the two buttons for the form sign-in modal.
* Fixed the issue of modals persisting across route changes within an app. One takeaway in my local testing is how awful Safari's handling back/forward makes this look no matter what...

> <img width="611" alt="screen shot 2019-03-04 at 8 35 21 pm" src="https://user-images.githubusercontent.com/1067024/53928850-9f46e000-4059-11e9-88a9-5a4710f80e73.png">

narin reviewed (approved):
LGTM. Tested locally and confirmed modal displays when trying to sign in during an incomplete form.
